### ISSUE TITLE
Feature request: add an `add subtitle` function for video playing

### ISSUE THREAD

**Issue by Sollovin** [pldubouilh/gossa#62]
Hello, as an nonnative English speaker, a function to add subtitles for the streaming videos would be very useful for me. Do you have any plan to implement such a functionality?

**Comment by pldubouilh** [comment]
Yes I'd like subtitles support as well - but sadly the [html videoplayer doesnt support .srt](https://developer.mozilla.org/en-US/docs/Web/Guide/Audio_and_video_delivery/Adding_captions_and_subtitles_to_HTML5_video) :(

There are 2 options: using a JS library to fetch the .srt and [mangle it to WebVTT client side](https://github.com/EldinZenderink/SubPlayerJS), or [mangle the subtitles server side](https://github.com/asticode/go-astisub) so that the videplayer can read them natively.

I'm not really happy with either solutions as they both rely on external dependencies. I've been considering making a separate project, a proxy to convert videos from random formats to mp4, to make them readable by browsers (including subtitles then), but I'm not really sure of the relevance at this stage.

**Comment by lakuapik** [comment]
Hi @pldubouilh, thanks for this awesome project! 

just asking, are you planning on adding native support for .vtt? i dont mind converting my srt to vtt manually, there seems plenty of online tools that can help to do that.

or do you still planning on adding .srt support just like what discussed above? many thanks.

**Comment by pldubouilh** [comment]
Hey @lakuapik, thanks for raising this issue again ! I'll re-open it and try think of solutions. 
Supporting .vtt sounds like a good start indeed, as it's supported natively. If you feel like writing a PR, that's also welcomed :)
